xdsl/backend/x86/lowering/helpers.py

```diff
@@ -39,8 +40,9 @@ def vector_type_to_register_type(
 
 def scalar_type_to_register_type(value_type: Attribute) -> type[RegisterType]:
     assert not isinstance(value_type, ShapedType)
-    assert isinstance(value_type, FixedBitwidthType)
-    if value_type.bitwidth <= 64:
+    if (
+        isinstance(value_type, FixedBitwidthType) and value_type.bitwidth <= 64
+    ) or isinstance(value_type, IndexType):
         return x86.register.GeneralRegisterType
     else:
         raise DiagnosticException("Not implemented for bitwidth larger than 64.")
```

Comment:
How about using `CompileTimeFixedBitwidthType` instead, and passing the bitwidth as a parameter? The usual name in MLIR is `index_bitwidth`, that way if the type is representable in less than the register size it gets put in a register.

Reply:
yeah, that way the 64 isn't hardcoded

Reply:
Yeah actually you're right.